varinfo/umm_var.py

```diff
@@ -346,53 +346,74 @@ def get_json_serializable_value(input_value: Any) -> Any:
 def publish_umm_var(collection_id: str,
                     umm_var_dict: Dict,
                     auth_header: str,
-                    cmr_env: CMR_UAT = CMR_UAT) -> Union[str, Dict]:
+                    cmr_env: CmrEnvType = CMR_UAT) -> str:
     """" Publish a single UMM-Var entry to CMR given:
         * collection_id: a collection's concept_id
         * umm_var_dict: a dictionary of a single UMM-Var entry for a collection
-        * auth_header: Earthdata Login (EDL) bearer_token or LaunchPad token
-            with their respective headers
+        * auth_header: Authorization HTTP header, containing a LaunchPad
+          token: 'Authorization: <token>'
         * cmr_env: CMR environments (OPS, UAT, and SIT) default is CMR_UAT
-    For a successful requests all of these fields must be entered
+
+        For a successful requests all of these fields must be entered
 
     """
     # Required UMM-Var headers for ingesting variable entries
     headers_umm_var = {
         'Content-type': 'application/vnd.nasa.cmr.umm+json;version='
         + f'{umm_var_dict["MetadataSpecification"]["Version"]}',
         'Authorization': auth_header,
-        'Accept': 'application/json'}
+        'Accept': 'application/json'
+    }
+    variable_native_id = generate_variable_native_id(collection_id,
+                                                     umm_var_dict)
 
     url_endpoint = (cmr_env.replace('search', 'ingest') + 'collections/'
-                    f'{collection_id}/variables/{umm_var_dict["Name"]}')
+                    f'{collection_id}/variables/{variable_native_id}')
```

Comment:
When I was testing the new function locally, I was getting a n error on publication where CMR couldn't find the endpoint we were looking for. Turns out it's because my test collection was hierarchical, so:

```
https://cmr.uat.earthdata.nasa.gov/ingest/collections/<collection concept ID>/variables/<native ID>
```

Looks something like:

```
https://cmr.uat.earthdata.nasa.gov/ingest/collections/C1234567890-PROV/variables/Grid/time
```

That extra slash was confusing CMR, making it thing we were looking for a different endpoint.

Reply:
Using the collection concept ID as part of the native ID is a bit less human-friendly, but it's the only collection information was available to the function to ensure uniqueness within the provider. Also, I think it's generally only metadata curators that tend to see/use the native ID.